### ISSUE TITLE
Travis tests are installing celery 4.1 (latest), but the README says latest tested version is celery 3.1.19

### ISSUE THREAD

**Issue by TysonAndre** [gjedeer/celery-php#109]
This issue refers to the master branch (3.0.0-dev)

Desired: Run travis tests with celery 3.1.x (e.g. 3.1.25), alongside other supported celery versions, clarify documentation. This can be done via the travis `env:` directive (E.g. `CELERY_INSTALLATION=celery==3.1.25` could be used to cause travis to run `pip install celery==3.1.25 redis`.

Observed: Celery 4.1.0 is installed in travis tests. From the documentation, I'm not clear if celery 4.1 is officially supported yet, and limitations aren't summarized.

- Also, latest builds are failing according to https://travis-ci.org/gjedeer/celery-php/branches (EDIT: never mind, PR 108 may have a fix)
- Also, I'm not able to get the unit tests to pass with celery 3.1.x (3.1.18 or 3.1.25). I created an equivalent dockerized test runner to run tests locally, for which tests pass in Celery 4.1.0, but many fail in 3.1.x (nothing else changed except dependencies of celery). (https://github.com/TysonAndre/celery-php/blob/master/run_docker_test.sh)

   Is celery 3.1 supported in the latest master commits?

Celery 4.1 installation can be seen in https://travis-ci.org/gjedeer/celery-php/jobs/318734521#L894

```sh
$ venv/bin/pip install celery redis
Collecting celery
/home/travis/build/gjedeer/celery-php/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:318: SNIMissingWarning: An HTTPS request has been made, but the SNI (Subject Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#snimissingwarning.
  SNIMissingWarning
/home/travis/build/gjedeer/celery-php/venv/local/lib/python2.7/site-packages/pip/_vendor/requests/packages/urllib3/util/ssl_.py:122: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/security.html#insecureplatformwarning.
  InsecurePlatformWarning
  Downloading celery-4.1.0-py2.py3-none-any.whl (400kB)
    100% |████████████████████████████████| 409kB 2.2MB/s 
Collecting redis
  Downloading redis-2.10.6-py2.py3-none-any.whl (64kB)
    100% |████████████████████████████████| 71kB 4.6MB/s 
Collecting billiard<3.6.0,>=3.5.0.2 (from celery)
  Downloading billiard-3.5.0.3.tar.gz (149kB)
    100% |████████████████████████████████| 153kB 4.1MB/s 
Collecting pytz>dev (from celery)
  Downloading pytz-2017.3-py2.py3-none-any.whl (511kB)
    100% |████████████████████████████████| 512kB 2.2MB/s 
Collecting kombu<5.0,>=4.0.2 (from celery)
  Downloading kombu-4.1.0-py2.py3-none-any.whl (181kB)
    100% |████████████████████████████████| 184kB 4.1MB/s 
Collecting amqp<3.0,>=2.1.4 (from kombu<5.0,>=4.0.2->celery)
  Downloading amqp-2.2.2-py2.py3-none-any.whl (48kB)
    100% |████████████████████████████████| 51kB 9.2MB/s 
Collecting vine>=1.1.3 (from amqp<3.0,>=2.1.4->kombu<5.0,>=4.0.2->celery)
  Downloading vine-1.1.4-py2.py3-none-any.whl
Building wheels for collected packages: billiard
  Running setup.py bdist_wheel for billiard ... done
  Stored in directory: /home/travis/.cache/pip/wheels/85/15/e4/11683b23ab74c2a835845811976e664ab33df7d23c3cb23500
Successfully built billiard
Installing collected packages: billiard, pytz, vine, amqp, kombu, celery, redis
Successfully installed amqp-2.2.2 billiard-3.5.0.3 celery-4.1.0 kombu-4.1.0 pytz-2017.3 redis-2.10.6 vine-1.1.4
```

**Comment by gjedeer** [comment]
Hi, yes, the master branch is still for celery < 4 (or, more specifically, protocol v1), the only reason why tests are installing celery 4 is because at the time I wrote .travis.yml, Celery 4 was still in beta (so `pip install celery` installed celery 3).

I'm not sure why your tests are failing. The last commit which I tested manually with Celery 3.1.19 is https://github.com/gjedeer/celery-php/commit/97a08bccc746d1cd6d34ce1823260eb7c7b3d969#diff-04c6e90faac2675aa89e2176d2eec7d8 so maybe try rolling back and using that version. I don't really use this library any more or spend time supporting it.
